File: leo2-common/src/main/java/org/deku/leo2/messaging/log/v1/LogMessage.java
package org.deku.leo2.messaging.log.v1;

import ch.qos.logback.classic.Level;
import ch.qos.logback.classic.spi.LoggingEvent;
import org.slf4j.helpers.MessageFormatter;

import java.io.Serializable;

/**
 * Created by masc on 16.04.15.
 */
public class LogMessage implements Serializable {
    private static final long serialVersionUID = -8027400236775552276L;

    private String mLevel;
    private String mLoggerName;
    private String mThreadName;
    private String mMessage;
    private long mTimestamp;

    public LogMessage() {
    }
    public LogMessage(LoggingEvent le) {
        mLevel = le.getLevel().toString();
        mLoggerName = le.getLoggerName();
        mThreadName = le.getThreadName();
        mMessage = (le.getArgumentArray() != null) ?
                MessageFormatter.arrayFormat(le.getMessage(), le.getArgumentArray()).getMessage() : le.getMessage();
        mTimestamp = le.getTimeStamp();
    }

    public String getLevel() {
        return mLevel;
    }

    public String getLoggerName() {
        return mLoggerName;
    }

    public String getThreadName() {
        return mThreadName;
    }

    public String getMessage() {
        return mMessage;
    }

    public long getTimestamp() {
        return mTimestamp;
    }
}

File: libs/leoz-common/src/main/java/org/deku/leoz/messaging/log/LogMessage.kt
package org.deku.leoz.messaging.log

import ch.qos.logback.classic.spi.LoggingEvent
import org.slf4j.helpers.MessageFormatter

import java.io.Serializable

/**
 * Log message
 * Created by masc on 16.04.15.
 */
public class LogMessage : Serializable {
    companion object {
        private val serialVersionUID = -8027400236775552276L
    }

    public var level: String = ""
    public var loggerName: String = ""
    public var threadName: String = ""
    public var message: String = ""
    public var timestamp: Long = 0

    public constructor() { }

    public constructor(le: LoggingEvent) {
        level = le.level.toString()
        loggerName = le.loggerName
        threadName = le.threadName
        message = if ((le.argumentArray != null))
            MessageFormatter.arrayFormat(le.message, le.argumentArray).message
        else
            le.message
        timestamp = le.timeStamp
    }
}
